token-lending/program/src/processor.rs

```diff
@@ -1696,6 +1701,52 @@ fn process_flash_loan(
     Ok(())
 }
 
+
+#[inline(never)] // avoid stack frame limit
+fn process_close_obligation_account(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let obligation_info = next_account_info(account_info_iter)?;
+    let obligation_owner_info = next_account_info(account_info_iter)?;
+    let destination_account_info = next_account_info(account_info_iter)?;
+
+    let obligation = Obligation::unpack(&obligation_info.data.borrow())?;
+    if &obligation.owner != obligation_owner_info.key {
+        msg!("Obligation owner does not match the obligation owner provided");
+        return Err(LendingError::InvalidObligationOwner.into());
+    }
+    if !&obligation.borrows.is_empty() {
+        msg!("Obligation borrows must be zero");
+        return Err(LendingError::NonZeroObligationBorrow.into());
+    }
+    if !obligation.deposits.is_empty() {
+        msg!("Obligation deposits must be zero");
+        return Err(LendingError::NonZeroObligationDeposit.into());
+    }
+    
+    if obligation_info.owner != program_id {
+        msg!("Obligation provided is not owned by the lending program");
+        return Err(LendingError::InvalidAccountOwner.into());
+    }
+    if !obligation_owner_info.is_signer {
+        msg!("Obligation owner provided must be a signer");
+        return Err(LendingError::InvalidSigner.into());
```

Comment:
nit: we could return a `ProgramError::MissingRequiredSignature` instead to be clearer

Reply:
I'll implement your suggestions. Thanks for rearing this pull-request!